chapter_convolutional-modern/resnet.md

```diff
@@ -384,63 +381,26 @@ with d2l.try_gpu():
+Different from the smorgasbord of transformations in Inception, 
 ResNeXt adopts the *same* transformation in all branches,
-thus minimizing manual design efforts in each branch.
+thus minimizing the need for manual tuning of each branch. 
 
-![The ResNeXt block. It is a bottleneck (when $b < c$) residual block with group convolution ($g$ groups).](../img/resnext-block.svg)
+![The ResNeXt block. The use of group convolution with $g$ groups is $g$ times faster than a dense convolution. It is a bottleneck residual block when the number of intermediate channels $b$ is less than $c$.](../img/resnext-block-simplified.svg)
 :label:`fig_resnext_block`
 
-The left dotted box in
-:numref:`fig_resnext_block`
-depicts the added concatenated parallel transformation
-strategy in ResNeXt.
-More concretely,
-an input with $c$ channels
-is first split into $g$ groups
-via $g$ branches of $1 \times 1$ convolutions
-followed by $3 \times 3$ convolutions,
-all with $b/g$ output channels.
-Concatenating these $g$ outputs
-results in $b$ output channels,
-leading to "bottlenecked" (when $b < c$) network width
-inside the dashed box.
-This output
-will restore the original $c$ channels of the input
-via the final $1 \times 1$ convolution
-right before sum with the residual connection.
-Notably,
-the left dotted box is equivalent to
-the much *simplified* right dotted box in :numref:`fig_resnext_block`,
-where we only need to specify
-that the $3 \times 3$ convolution is a *group convolution*
-with $g$ groups.
-In fact,
-the group convolution dates back
-to the idea of distributing the AlexNet
-model over two GPUs due to limited GPU memory at that time :cite:`Krizhevsky.Sutskever.Hinton.2012`.
-
-The following implementation of the `ResNeXtBlock` class
-treats `groups` ($b/g$ in :numref:`fig_resnext_block`) as an argument
-so that given `bot_channels` ($b$ in :numref:`fig_resnext_block`) bottleneck channels,
-the $3 \times 3$ group convolution will
-have `bot_channels//groups` groups.
-Similar to
-the residual block implementation in
-:numref:`subsec_residual-blks`,
-the residual connection
-is generalized
-with a $1 \times 1$ convolution (`conv4`),
-where setting `use_1x1conv=True, strides=2`
-halves the input height and width.
+Breaking up a convolution from $c$ to $b$ channels into one of $g$ groups of size $c/g$ generating $g$ outputs of size $b/g$ is called, quite fittingly, a *grouped convolution*. The computational cost is reduced from $O(c \cdot b)$ to $O(g \cdot (c/g) \cdot (b/g)) = O(c \cdot b / g)$, i.e. it is $g$ times faster. Even better, the number of parameters needed to generate the output is also reduced from a $c \times b$ matrix to $g$ smaller matrices of size $(c/g) \times (b/g)$, again a $g$ times reduction. In what follows we assume that both $b$ and $c$ are divisible by $g$. 
```

Comment:
According to Fig 3(b) of [the ResNeXt paper](https://openaccess.thecvf.com/content_cvpr_2017/papers/Xie_Aggregated_Residual_Transformations_CVPR_2017_paper.pdf), the input channel $c$ (256 in Fig 3(b)) remains the same (rather than being divided into $c/g$ channels) in every group. Thus,

"Breaking up a convolution from $c$ to $b$ channels into one of $g$ groups of size $c/g$ generating $g$ outputs of size $b/g$"
->
"Breaking up a convolution from $c$ to $b$ channels into one of $g$ groups of size $c$ generating $g$ outputs of size $b/g$"

Since this affects the subsequent arguments, could you please double check and revise if necessary? Please correct me if I am wrong. Thanks.

Reply:
No, your interpretation isn't correct. The point is that you have an initial bottleneck to go from c (channel) to b (bottleneck) dimensions. After that, each grouped convolution (there are g groups) acts only on a small slice of b/g inputs and produces b/g outputs. Otherwise there wouldn't be any benefit in using ResNeXt.

Reply:
Agreed. So $c$ and $b$ in text should be $c_{in}$ and $c_{out}$ in general, not to be confused with $c$ and $b$ in Fig.8.6.5. Actually for the grouped convolution in Fig.8.6.5, $c_{in} = c_{out} = b$.